<?xml version="1.0"?>

<launch>
  <!-- Run the map server -->
  <arg name="map_file" default="$(find footstep_planner)/maps/RCGO2011_Hector_Mission4_2.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" >
    <param name="frame_id" value="/world" />
  </node>

  <!-- Run terrain classifier -->
  <include file="$(find flor_terrain_classifier)/launch/terrain_classifier_test.launch" />

  <!-- Run basic footstep planner of ros -->
  <!-- launch-prefix="valgrind -/-tool=callgrind" -->
  <node name="ros_footstep_planner" pkg="footstep_planner" type="footstep_planner_node" output="screen">
    <rosparam file="$(find footstep_planner)/config/ros_footstep_planner_params.yaml" command="load" />
    <rosparam file="$(find footstep_planner)/config/footsteps_atlas.yaml" command="load" />
    <rosparam file="$(find footstep_planner)/config/atlas_params.yaml" command="load" />
    <param name="gpr_file" value="$(find footstep_planner)/config/standard.gpr" />
    <param name="map_step_cost_file" value="$(find footstep_planner)/config/model_reduced.map" />
    <!-- <remap from="ground_level_grid_map" to="/flor/worldmodel/ground_level_grid_map" />  -->
    <!-- <remap from="body_level_grid_map" to="/flor/worldmodel/body_level_grid_map" />  -->
    <remap from="ground_level_grid_map" to="map" />
    <remap from="body_level_grid_map" to="map" />
    <remap from="initialpose" to="start_pose" />
    <remap from="goal" to="goal_pose" />
  </node>
  
  <!-- This launches the transformer node that takes in the plan, converts it into the bdi frame and sends it to the controller-->
  <include file="$(find flor_footstep_plan_transformer)/launch/footstep_plan_transformer_drc_default.launch" />
</launch>
